### PR TITLE
[picotts] Add LRU cache

### PR DESCRIPTION
**Pull request by dalgwen**
This pull request make use of the [new TTS LRU cache](https://github.com/openhab/openhab-core/pull/3057) by applying it to the Pico TTS service.

### CODE REVIEW

File: bundles/org.openhab.voice.picotts/src/main/java/org/openhab/voice/picotts/internal/PicoTTSService.java [hunk 1]
```diff
@@ -23,17 +23,27 @@
...
 import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Florian Schmidt - Initial Contribution
  */
 @Component
 @NonNullByDefault
-public class PicoTTSService implements TTSService {
+public class PicoTTSService extends AbstractCachedTTSService implements TTSService {
```
Comment (lolodomo) [inline bundles/org.openhab.voice.picotts/src/main/java/org/openhab/voice/picotts/internal/PicoTTSService.java:40]: `AbstractCachedTTSService` already implements `TTSService `.
Another option is to rather use 
```
@Component(service = TTSService.class)
```

**Review by lolodomo — COMMENTED** [review]
Just one comment.

**Comment by lolodomo** [comment]
No risk of problem with the audio servlet for this one ?

**Comment by lolodomo** [comment]
> No risk of problem with the audio servlet for this one ?

I answer to myself: this TTS service always create a (temporary) file, so no risk.

**Comment by lolodomo** [comment]
picotts (like mactts) is creating a temporary file.
I think we could add a method in `AbstractCachedTTSService` that would be called by the cache mechanism when the cache  has created its own file from the returned AudioStream. That method will inform the TTS service that it can release/delete its file. This will avoid having audio files duplicated, one as a temporary file by the TTS service and one in the TTS cache.

WDYT ?

**Comment by dalgwen** [comment]
Yes, indeed, it is not clean to keep several files.
If I had to do this, I would create an interface Releasable (or something like this), with a method release() inside.
The FileAudioStream, PicoTTSAudioStream, etc, could then implements it.

And inside LRUMediaCacheEntry, in the read method (line 281, which is the only place where we could say with certainty that the original InputStream is no longer needed), we could check if the InputStream implements the Releasable interface and in this case call the release method (which, in our case, is a deletion).

**Comment by lolodomo** [comment]
We could even make it simpler. Just add a method `release` to `FileAudioStream`. That would also require to fix the picotts and mactts so that their audio stream class extend FileAudioStream instead of FixedLengthAudioStream. Would it be not enough "clean" to do that ?

**Comment by lolodomo** [comment]
In the addons database, the only bindings really using FileAudioStream (other than for its public constants) are 3 TTS services: mimic, voicerss and polytts.
So if we change `FileAudioStream`, it will be easy to check the possible impact on the existing addons suing it.

**Comment by lolodomo** [comment]
Regarding the file deletion, I believe we should let each TTS service decide what to do with the files they produced.
So I would really prefer a new method in `AbstractCachedTTSService` that will be called by the TTS cxache handler.

**Comment by lolodomo** [comment]
I now realized that the place where it should be called is in LRUMediaCacheEntry and we don't have at this place an access to the `AbstractCachedTTSService`.

> If I had to do this, I would create an interface Releasable (or something like this), with a method release() inside.
> The FileAudioStream, PicoTTSAudioStream, etc, could then implements it.

Yes, I think now that that you're right.

**Comment by lolodomo** [comment]
@dalgwen : I believe you just have to add the `dispose` method like I did in #15109

**Comment by lolodomo** [comment]
> @dalgwen : I believe you just have to add the `dispose` method like I did in #15109

Can you add that and look at my comment ?
Then I will approve and merge your PR.

**Comment by lolodomo** [comment]
@dalgwen : no chance to finish? 
If you prefer, I can create an alternative PR.

**Comment by dalgwen** [comment]
@lolodomo 
Excuse me I forgot this, was thinking about finishing sinks first. I will check this this today or tomorrow.

**Comment by dalgwen** [comment]
dispose and readme done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by lolodomo** [comment]
I merged this PR immediately because a `FixedLengthAudioStream` is used by this TTS service, so we do not depend on any enhancement of audio sinks.

**Pull request merged**